dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/ClientWithNoCdi.java

```diff
@@ -65,26 +69,40 @@ public static void setUp() throws Exception {
                         .addPackage(ClientTriggeringServlet.class.getPackage());
         CDIArchiveHelper.addBeansXML(noCDIApp, com.ibm.websphere.simplicity.beansxml.BeansAsset.DiscoveryMode.NONE);//Not strictly needed unless we expand this test to cover MP rest client.
 
-
         ShrinkHelper.exportAppToServer(server, noCDIApp, SERVER_ONLY);
         server.startServer();
     }
 
     //This test does not use open telemetry directly, but it does trigger io.openliberty.microprofile.telemetry.internal.rest.TelemetryClientFilter.java for an app with no CDI and ensures it does not throw an exception
+
+    //This test also ensures that we actually call the started and shutdown methods in OpenTelemetryInfoFactoryImpl
+    //I'm putting it in here to avoid creating an extra server that enables trace. And in an existing method because
+    //Our version of junit doesn't allow fixed test ordering.
     @Test
-    public void testClientWithNoCDi() throws Exception {
+    public void testClientWithNoCDIAndFactoryStarupShutdownCalled() throws Exception {
         HttpRequest httpRequest = new HttpRequest(server, "/" + NO_CDI_APP_NAME + "/ClientTriggeringServlet");
         assertEquals(io.openliberty.microprofile.telemetry.internal_fat.apps.clientnocdi.ClientInvokedServlet.TEST_PASSED, httpRequest.run(String.class));
 
-        assertFalse("The test did not use a TelemetryClientFilter object", 
-                     server.findStringsInLogsUsingMark("microprofile.telemetry\\d*.internal.rest.TelemetryClientFilter < isEnabled Exit", server.getDefaultTraceFile())
-                     .isEmpty());
+        assertNotNull("The test did not use a TelemetryClientFilter object",
+                      server.waitForStringInLog("microprofile.telemetry\\d*.internal.rest.TelemetryClientFilter < isEnabled Exit", server.getDefaultTraceFile()));
+
+        server.deleteDirectoryFromLibertyServerRoot("apps");
+
+        assertNotNull("OpenTelemetryInfoFactoryImpl.applicationStarting was not invoked",
+                      server.waitForStringInLog("OpenTelemetryInfoFactoryImpl > applicationStarting Entry", server.getDefaultTraceFile()));
+
+        assertNotNull("OpenTelemetryInfoFactoryImpl.applicationStopped was not invoked",
+                      server.waitForStringInLog("OpenTelemetryInfoFactoryImpl > applicationStopped", server.getDefaultTraceFile()));
 
+        assertNotNull("OpenTelemetryInfoImpl was not disposed",
+                      server.waitForStringInLog("EnabledOpenTelemetryInfo > dispose Entry", server.getDefaultTraceFile()));
     }
```

Comment:
As a follow-up:

1. I don't like the way this test class sets up the app in the setup method, and then deletes the app in the test method. If you need to test something that happens when an app is removed, I think it makes more sense to deploy and then remove the app within the test method.

   Mostly, our tests deploy apps at startup and they remain running until the server is shut down. This test half-conforms to that pattern. It looks like a regular test but has a subtle line in the middle which undeploys the app. It would be easy for someone to add a new test to this class that may well work initially, depending on the execution order, but will fail if the ordering is different.

2. Don't use one test to test two things. You can use the same server, but if you need to test an app being removed, please use a separate app and deploy and remove it within the test method.

Reply:
For 2 I did it this way to avoid creating an extra server with trace.log enabled, but if you prefer it I could move it to a new test without trouble.